### ISSUE TITLE
`InvisibleButton`: add a way to optionally call `RenderNavHighlight` inside?

### ISSUE THREAD

**Issue by achabense** [ocornut/imgui#8057]
### Version/Branch of Dear ImGui:

Version 1.91.2

### Back-ends:

imgui_impl_sdl2.cpp + imgui_impl_sdlrenderer2.cpp

### Compiler, OS:

Windows 10 + MSVC 2022

### Details:

`ImGui::InVisibleButton` is useful for defining custom widgets. However, when `ImGuiConfigFlags_NavEnableKeyboard` is specified, the keyboard navigation will work on them in an invisible way.
Take the code below for example. If not calling `RenderNavHighlight`, the effect would be like this. It's hard to tell where the navigation goes.
<img width="300" alt="Example" src="https://github.com/user-attachments/assets/9ea53ef2-8944-4fe8-9424-059f8f29c276">
It will work well if calling `RenderNavHighlight`. However, as `RenderNavHighlight` is an internal function, I think it will be helpful to add a way to call `RenderNavHighlight` inside `InvisibleButton`. A new button flag may be fine for this change.
<img width="300" alt="Example" src="https://github.com/user-attachments/assets/ef7b159b-f6fd-4bb8-9426-7df17aa24a81">

### Minimal, Complete and Verifiable Example code:

```cpp
// ImGui::GetIO().ConfigFlags |= ImGuiConfigFlags_NavEnableKeyboard;

static void example() {
    if (ImGui::Begin("Example", 0, ImGuiWindowFlags_NoSavedSettings)) {
        const int c = 5;
        static bool vals[c]{};
        for (int i = 0; i < c; ++i) {
            if (i != 0) {
                ImGui::SameLine();
            }
            ImGui::PushID(i);
            if (ImGui::InvisibleButton("In", {20, 20})) {
                vals[i] = !vals[i];
            }
            ImGui::PopID();

            // Custom rendering:
            const auto min = ImGui::GetItemRectMin();
            const auto max = ImGui::GetItemRectMax();
            // ImGui::RenderNavHighlight({min, max}, ImGui::GetItemID()); // "imgui_internal.h"
            ImGui::GetWindowDrawList()->AddRectFilled(min, max, vals[i] ? IM_COL32_WHITE : IM_COL32_BLACK);
            ImGui::GetWindowDrawList()->AddRect(min, max, IM_COL32_WHITE);
        }
    }
    ImGui::End();
}
```

**Comment by ocornut** [comment]
A convenient default may to do the opposite:
- Change `InvisibleButton()` to use _NoNav by default and so it is correct that it doesn't render highlight. (we need to bundle those two properties together to be consistent)
- And you can opt-out aka re-enable navigation.
It's a little bit inconsistent but no breakage and might be a better default. I will be polling a few large users.

**Comment by ocornut** [comment]
I have been agonizing a little too much over this...
It would be consistent to enable nav by default, but with the fix of rendering highlight this is going to create many more issues (e.g. ctrl+tabbing back to a window with a large invisible button).

I also polled some users, and as one in particular said "we have 595 calls to InvisibleButton() in our codebase" I though I would play it safe 😆 

Pushed 67e5f35, disable nav by default + added `ImGuiButtonFlags_EnableNav`.

There is a possibility that someone would want to use `InvisibleButton()` with nav but no highlight (drawing a custom one). In which case we could decide to specifically add a `ImGuiItemFlags_NoNavHighlight` item flag. But I suspect most people are going to use lower-level `ButtonBehavior()` in this situation anyhow.
